### PR TITLE
Enable package extension for Statistics

### PR DESCRIPTION
**Pull request by hyrodium**
This PR reverts #255.

### CODE REVIEW

**Comment by kimikage** [comment]
This seems more elegant than the use of Requires.jl with respect to Stdlib.

On the other hand, in the first place, this workaround uses a non-public function `Statistics._mean_promote`.
Until now, this has been acceptable because the version of Statistics.jl was synchronized with Julia's version upgrades.

Seems to me we need to revisit the workaround itself.
Of course, that should be separate from this PR.

**Comment by kimikage** [comment]
This change is not so complicated, and the intent is clear.
If we decide to merge this PR, however, it might be easier to track if the simple `revert` commit was made in master.
This is just a little reminder for the hygiene of commit-graph, because PR #278 also makes changes to Project.toml.

**Comment by johnnychen94** [comment]
Sorry, I missed this PR.

For the very fundamental packages such as FixedPointNumbers, keeping a broad range of compatibility is necessary. FixedPointNumbers is still compatible with Julia 1.0, which is great.

Thus, we still need to preserve the old codes, still need to install Requires (but conditionally loading it depending on the actual Julia versions).

Ref: https://pkgdocs.julialang.org/v1/creating-packages/#Backwards-compatibility

**Comment by kimikage** [comment]
@johnnychen94 You are absolutely right.
We might need Requrires.jl (perhaps for CheckedArithmetic?).

But for the case of Statistics, it has been a stdlib and should work as before. (Conversely, the "new” Statistics.jl basically won't be introduced into the old julia.)

**Comment by johnnychen94** [comment]
> Adding extension support is a good change, but it requires Julia 1.9 to work.

Sorry, I misunderstand the code. The current version **_is_** compatible with old Julia versions by falling back to a direct `include` rather than the lazy `@require`.

My two cents is to still use the lazy `@require` because it's what we expected in #255.

File: src/FixedPointNumbers.jl
Comment (kimikage) [inline src/FixedPointNumbers.jl:628]: Although this is in keeping with PR #255 manner, I would prefer that the `precompile` be at the end of the module (except for the `__init__()`).

**Comment by kimikage** [comment]
@hyrodium Could you `rebase` (and tweak) this?

File: Project.toml
Comment (kimikage) [inline Project.toml:24]: I don't fully understand how Package Extension works, but this package still hardly `[dep]`ends on `Statistics` for the time being, so I guess it is not needed for `[extras]` and `test`.

**Review by johnnychen94 — APPROVED** [review]

File: Project.toml
Comment (kimikage) [inline Project.toml:16]: If we follow the principle, `Statistics` should be in `[compat]`.
This is especially the case where even patch versions need to be included (e.g. `Statistics = "1 - 1.11.1"`).

**Review by kimikage — COMMENTED** [review]
It is hard to determine where in the trade-off we should compromise.

**Comment by kimikage** [comment]
cf. https://github.com/JuliaStats/Statistics.jl/issues/165

**Comment by kimikage** [comment]
@hyrodium Thank you for rebasing.
Is this ready to merge? Or do you plan to revise it (regarding my comments)?
I will merge this PR once I get the go-ahead.

File: Project.toml
Comment (hyrodium) [reply Project.toml]: Fixed in https://github.com/JuliaMath/FixedPointNumbers.jl/pull/277/commits/6c2a68cc9513d092be75289e840daac58fc3c2e0.

File: src/FixedPointNumbers.jl
Comment (hyrodium) [reply src/FixedPointNumbers.jl]: Fixed in https://github.com/JuliaMath/FixedPointNumbers.jl/pull/277/commits/c4eee40f2aa33258e150259de49468bf73c20838.

Comment (hyrodium) [reply src/FixedPointNumbers.jl]: Actually, I'm not familiar with `_precompile_()` function. Perhaps the current code should be replaced with [PrecompileTools.jl](https://github.com/JuliaLang/PrecompileTools.jl).

File: Project.toml
Comment (kimikage) [reply Project.toml]: To my surprise, it seems that the hyphen notation is not supported for julia v1.0.
Sorry for misleading you.

File: src/FixedPointNumbers.jl
Comment (kimikage) [reply src/FixedPointNumbers.jl]: Yes, it is from the SnoopCompile.jl days.
I think replacing before releasing the new version is a good idea.

File: Project.toml
Comment (hyrodium) [reply Project.toml]: Yeah, `< 1.11.2` could be a solution (cf. [Pkg.jl v1.0 documentation](https://pkgdocs.julialang.org/v1.0/#Inequality-specifiers-1)). However, `"1 - 1.11.1"` looks better to me because it has a lower bound. Is there any chance to drop support for Julia 1.0?

Comment (kimikage) [reply Project.toml]: Since Statistics.jl is stdlib, the lower bound is practically meaningless.
I don't intend to continue actively supporting v1.0, but I don't think it has caused any critical problems yet.

Comment (kimikage) [reply Project.toml]: Also, this is very trivial, the uppercase "S" of Statistics precedes the lowercase "j" of julia.

Comment (hyrodium) [reply Project.toml]: Thanks for the comments! I have updated the compat bound for Statistics.jl

**Comment by kimikage** [comment]
@hyrodium Thank you for addressing this issue!

**Pull request merged**
